custom_components/easee/switch.py

```diff
@@ -24,26 +26,36 @@ class ChargerSwitch(ChargerEntity, SwitchEntity):
     async def async_turn_on(self, **kwargs):  # pylint: disable=unused-argument
         """Turn on the switch."""
         _LOGGER.debug("%s Switch turn on" % self._entity_name)
-        self.set_value_from_key(self._state_key, True)
-        self._state = True
-        self.async_write_ha_state()
         function_call = getattr(self.data.product, self._switch_func)
         try:
             await function_call(True)
+        except ForbiddenServiceException:
+            raise HomeAssistantError(
+                f"Forbidden {self._entity_name} turn_on - No access right"
+            ) from None
```

Comment:
If you add "return" after this, would that not solve that the switch changes state?